ament_cmake_core/cmake/symlink_install/ament_cmake_symlink_install_targets.cmake

```diff
@@ -89,8 +85,19 @@ function(ament_cmake_symlink_install_targets)
     string(REPLACE "\"LIBRARY\" \"DESTINATION\"" "\"LIBRARY_DESTINATION\"" argn_quoted "${argn_quoted}")
     string(REPLACE "\"RUNTIME\" \"DESTINATION\"" "\"RUNTIME_DESTINATION\"" argn_quoted "${argn_quoted}")
 
+    # generate unique files
+    set(generated_file
+      "${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_symlink_install_targets_${__AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX}.cmake")
+    math(EXPR __AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX
+      "${__AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX} + 1")
+    set(__AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX "${__AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX}"
+      CACHE INTERNAL "Index for unique symlink install targets")
```

Comment:
No, the purpose of the index is to generate unique files. At install time all of these files are being used / included. The state of the CMake cache is irrelevant because the variable is always initialized with a fixed value (https://github.com/ament/ament_cmake/pull/72/files/ea36f5eb6cf805644af6f9c2121d9fe26639d646#diff-1dc4e15369f0a9006d237086d487a774R15). So on every CMake invocation the index starts from zero again.